tests/shared/permissions.spec.js

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright 2017 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const m = require('mochainon');
+const os = require('os');
+const permissions = require('../../lib/shared/permissions');
+
+describe('Shared: permissions', function() {
+
+  describe('.getEnvironmentCommandPrefix()', function() {
+
+    describe('given windows', function() {
+
+      beforeEach(function() {
+        this.osPlatformStub = m.sinon.stub(os, 'platform');
+        this.osPlatformStub.returns('win32');
+      });
+
+      afterEach(function() {
+        this.osPlatformStub.restore();
+      });
+
+      it('should return an empty array if no environment', function() {
+        m.chai.expect(permissions.getEnvironmentCommandPrefix()).to.deep.equal([]);
+      });
+
+      it('should return an empty array environment is an empty object', function() {
+        m.chai.expect(permissions.getEnvironmentCommandPrefix({})).to.deep.equal([]);
+      });
+
+      it('should create an environment command prefix out of one variable', function() {
+        m.chai.expect(permissions.getEnvironmentCommandPrefix({
+          FOO: 'bar'
+        })).to.deep.equal([
+          'set',
+          'FOO=bar',
+          '&&',
+          'call'
+        ]);
+      });
+
+      it('should create an environment command prefix out of many variables', function() {
+        m.chai.expect(permissions.getEnvironmentCommandPrefix({
+          FOO: 'bar',
+          BAR: 'baz',
+          BAZ: 'qux'
+        })).to.deep.equal([
+          'set',
+          'FOO=bar',
+          '&&',
+          'set',
+          'BAR=baz',
+          '&&',
+          'set',
+          'BAZ=qux',
+          '&&',
+          'call'
+        ]);
+      });
+
+      it('should ignore undefined and null variable values', function() {
+        m.chai.expect(permissions.getEnvironmentCommandPrefix({
+          FOO: null,
+          BAR: undefined,
+          BAZ: 'qux'
```

Comment:
Only a minor nit, but for unit-testing purposes perhaps you should rearrange the values as `null, 'qux', undefined` which checks that valid-values work after null, and also checks that undefined values are ignored _after_ a valid value?